clients/signin/signin.go

```diff
@@ -0,0 +1,56 @@
+package signin
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"syscall"
+
+	"github.com/influxdata/influx-cli/v2/api"
+	"golang.org/x/term"
+)
+
+func GetCookie(ctx context.Context, params api.ConfigParams, userPass string) (string, error) {
+	bufUserPass, err := base64.StdEncoding.DecodeString(userPass)
+	if err != nil {
+		return "", err
+	}
+
+	userPass = strings.Trim(string(bufUserPass), "\x00")
+	splitUserPass := strings.Split(userPass, ":")
+	if len(splitUserPass) < 1 {
+		return "", fmt.Errorf("bad config")
+	}
+	username := splitUserPass[0]
+	var password string
+	if len(splitUserPass) != 2 {
+		fmt.Print("Please provide your password: ")
+		bytePassword, err := term.ReadPassword(int(syscall.Stdin))
+		if err != nil {
+			return "", err
+		}
+		password = string(bytePassword)
+		fmt.Println()
+	} else {
+		password = splitUserPass[1]
+	}
+
+	cfg := api.NewAPIConfig(params)
+	client := api.NewAPIClient(cfg)
+	ctx = context.WithValue(ctx, api.ContextBasicAuth, api.BasicAuth{
+		UserName: username,
+		Password: password,
+	})
+	res, err := client.SigninApi.PostSignin(ctx).ExecuteWithHttpInfo()
+	if err != nil {
+		return "", err
```

Comment:
Maybe if an error happens here, you could again print to the user that this is only supported for OSS, in case that clarifies their problem. What does the error look like when you try to make this request to cloud?

Reply:
It simply 404s because cloud tries to do some redirect stuff that we don't follow. I can wrap this error with a suggestion to make sure its not running against cloud.